src/azure-cli/azure/cli/command_modules/role/_help.py

```diff
@@ -353,6 +353,9 @@
 helps['ad sp create-for-rbac'] = """
 type: command
 short-summary: Create a service principal and configure its access to Azure resources.
+long-summary: >
+    When --skip-assignment is specified, --scopes will be ignored. You may use `az role assignment create` to create
+    role assignments for this service principal later.
```

Comment:
why not add this message in --skip-assignment parameter

Reply:
Because I think this message involves 2 parameters `--skip-assignment` and `--scopes`. In my mind, it is better to keep the message in a common place.

Reply:
I think you can add this message in `--skip-assignment` because it only impact when user set this parameter

---
In reply to: [441289495](https://github.com/Azure/azure-cli/pull/13977#discussion_r441289495) [](ancestors = 441289495)

Reply:
In such case, the help for `--scopes` won't indicate it will be ignored by `--skip-assignment`. It will be clearer to put cross-parameter help message in one place. Besides, it appears in `long-summary` and saves the help from being crowded.

Reply:
My personal experience is that I prefer to put parameter related information in the parameter help message. Because sometimes there may be too many parameters and we can't display all them in one screen, the user may not notice this information when he try to check the parameter specific message.

If it is a common pattern which related lots of parameters and will display ugly in each parameter, I think it's OK to put it in long summary.

For this case, I prefer to put this message in `--skip-assignment`. Sorry for not realize this before.

Reply:
To summarize, the debate is **where to put a help message involving multiple parameters**:

> When --skip-assignment is specified, --scopes will be ignored. You may use `az role assignment create` to create role assignments for this service principal later.

Reply:
Option 1: Put the help message in command help's `long-summary` section

Reply:
Option 2: Put the help message in any of the parameter, like `--skip-assignment` in this case, or both `--skip-assignment` and `--scope`.

Reply:
Do not use long summary. It is not shown in CLI reference. And docs teams does not plan to fix it.

Reply:
Put it in both places.

Reply:
@qwordy, this is tricky, and makes the `long-summary` totally broken doc-wise. 

Besides, I already applied option 1 in `az rest`:

```
> az rest -h

Command
    az rest : Invoke a custom request.
        This command automatically authenticates using the logged-in credential: If Authorization
        header is not set, it attaches header `Authorization: Bearer <token>`, where `<token>` is
        retrieved from AAD. The target resource of the token is derived from --url if --url starts
        with an endpoint from `az cloud show --query endpoints`. You may also use --resource for a
        custom resource.
        If Content-Type header is not set and --body is a valid JSON string, Content-Type header
        will default to application/json.
```

Reply:
> Another option: Put it in both places.

This introduces duplication, which IMO is bad.

Reply:
Moved the message to `--skip-assignment`:

```
> az ad sp create-for-rbac -h

Arguments
    ...
    --skip-assignment  : Skip creating the default assignment, which allows the service principal to
                         access resources under the current subscription. When specified, --scopes
                         will be ignored. You may use `az role assignment create` to create role
                         assignments for this service principal later.  Allowed values: false, true.
```